Adyen/UI/View Controllers/Secured View Controller/SecuredViewController.swift

```diff
@@ -105,54 +97,42 @@ public final class SecuredViewController: UIViewController {
-                                                                                object: nil,
-                                                                                queue: OperationQueue.main) { [weak self] _ in
-            self?.removeBlur()
-        }
-        
-        backgroundObservers = [toBackgroundObserver, backFromBackgroundObserver]
+        var array = [Any]()
+        array.append(notificationCenter.addObserver(forName: UIApplication.willResignActiveNotification,
+                                                    object: nil,
+                                                    queue: OperationQueue.main,
+                                                    using: { [weak self] _ in self?.addBlur() }))
+        array.append(notificationCenter.addObserver(forName: UIApplication.didBecomeActiveNotification,
+                                                    object: nil,
+                                                    queue: OperationQueue.main,
+                                                    using: { [weak self] _ in self?.removeBlur() }))
+        backgroundObservers = array
     }
     
     private func addBlur() {
         view.addSubview(blurEffectView)
-        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
         view.backgroundColor = .clear
         
-        let constraints = [
-            blurEffectView.leftAnchor.constraint(equalTo: view.leftAnchor),
-            blurEffectView.topAnchor.constraint(equalTo: view.topAnchor),
-            blurEffectView.rightAnchor.constraint(equalTo: view.rightAnchor),
-            blurEffectView.bottomAnchor.constraint(equalTo: view.bottomAnchor)
-        ]
-        
-        blurConstraints = constraints
-        
-        NSLayoutConstraint.activate(constraints)
+        blurEffectView.translatesAutoresizingMaskIntoConstraints = false
+        blurConstraints = blurEffectView.adyen.anchore(inside: view)
         
-        UIView.animate(withDuration: 0.2, animations: {
-            self.blurEffectView.effect = self.blurEffect
+        UIView.animate(withDuration: 0.2, animations: { [weak self] in
+            self?.blurEffectView.effect = self?.blurEffect
         })
     }
     
     private func removeBlur() {
-        UIView.animate(withDuration: 0.2, animations: {
-            self.blurEffectView.effect = nil
-        }, completion: { _ in
-            if let blurConstraints = self.blurConstraints {
-                NSLayoutConstraint.deactivate(blurConstraints)
-            }
-            self.blurEffectView.removeFromSuperview()
-            self.$blurEffectView.reset()
-            
-            self.view.backgroundColor = self.style.backgroundColor
-        })
+        UIView.animate(withDuration: 0.2,
+                       animations: { [weak self] in self?.blurEffectView.effect = nil },
+                       completion: { [weak self] _ in
```

Comment:
in places where there is a lot of `[weak self]` checks sometimes it is better to check whether `self` is `.some` at the beginning of the closure, so we don't need to execute it at all.

Reply:
Indeed.
We usually use `guard let self = self else { return }`